### PR TITLE
Adding attribution to GMCQ component

### PR DESCRIPTION
**Pull request by davetaz**
Ensuring that graphics can have attribution as per adapt-contrib-graphic

### CODE REVIEW

**Comment by davetaz** [comment]
Closes adaptlearning/adapt_framework#1528

File: templates/gmcq.hbs [hunk 1]
```diff
@@ -7,6 +7,9 @@
             <label aria-hidden="true" id="{{../_id}}-{{@index}}-aria" for="{{../_id}}-{{@index}}" class="{{#unless ../_isEnabled}}disabled {{/unless}} component-item-color component-item-text-color component-item-border {{#if _isSelected}}selected{{/if}} a11y-ignore" tabindex="-1">
 
                 <img src="{{_graphic.src}}" data-large="{{_graphic.large}}" data-small="{{_graphic.small}}"/>
+                {{#if _graphic.attribution}}
+                    <div class="gmcq-item-graphic-attribution">{{{_graphic.attribution}}}</div>
```
Comment (tomgreenfield) [inline templates/gmcq.hbs]: Would prefer to have a generic class on attributions for ease of override styling, i.e.
```html
<div class="graphic-attribution">{{{_graphic.attribution}}}</div>
```
You could still implement component-specific styles if necessary by using selectors like
```less
.gmcq .graphic-attribution {
    // styles
}
```

File: less/gmcq.less [hunk 2]
```diff
@@ -179,6 +179,11 @@
         }
     }
 
+    .gmcq-item-graphic-attribution {
+        font-size:0.75em;
+        line-height: 1em;
```
Comment (tomgreenfield) [inline less/gmcq.less]: @lc-thomasberger & @kirsty-hames, are we alright to introduce `em`s as a unit in core plugins? Can only recall seeing pixel values previously.

Comment (davetaz) [inline less/gmcq.less]: @tomgreenfield This style is an exact copy of the attribution code already in adapt-contrib-graphic/less/graphic.less. Happy to change, but I thought i'd keep it consistent to something already in the code base.

File: templates/gmcq.hbs (hunk 1 above)
Comment (davetaz) [inline templates/gmcq.hbs]: Happy to do this, I though it best this way, but @oliverfoster thought the other way. See adaptlearning/adapt-contrib-accordion/pull/81

Comment (davetaz) [inline templates/gmcq.hbs]: or maybe I misunderstood the atomic aspect of the comment...

Comment (oliverfoster) [inline templates/gmcq.hbs]: tom has a fair point. originally you had a generic style in the theme. i wanted a copy of the style in each plugin as that's where it's used.

you've renamed the class in such a way that to override all of the attribution divs from the theme you'd have to list them by plugin name.

tom's suggest into to make all the class names the same but nested inside their relevant containers.

Comment (davetaz) [inline templates/gmcq.hbs]: yes and that sounds more logical to me too. My bad for not interpreting that in that way. Will correct across all the PRs

Comment (oliverfoster) [inline templates/gmcq.hbs]: he's just corrected me somewhat. he wants

```
.mcq {

}
.graphic-attribution {
}
```

so that it's even less nestier.

Comment (tomgreenfield) [inline templates/gmcq.hbs]: Not too bothered either way; really depends how easy we'd like to override it in one fail sweep in the theme.

More nesting in plugin = more nesting in theme.

**Comment by davetaz** [comment]
@oliverfoster like that?

**Comment by oliverfoster** [comment]
i guess? @tomgreenfield ?

**Comment by tomgreenfield** [comment]
The `.graphic-attribution` selector *does* look lonely outside of the component nesting, not to mention there'll be several repeated global styles depending on how many graphic-attributed components are included.

I'd really like a third opinion from the FEDs. To clarify, the choice is described [here](https://github.com/adaptlearning/adapt-contrib-gmcq/pull/108#discussion_r112205061).

**Comment by oliverfoster** [comment]
ok. then we'll go for:
```
.mcq {
  .graphic-attribution {
    // styles
  }
}
```

keep it simple. sorry dave.

**Comment by davetaz** [comment]
.mcq or .gmcq-componenet? 
I don't see where .mcq is used in the component?

**Comment by oliverfoster** [comment]
sorry, my fault. too much code in one day

**Comment by oliverfoster** [comment]
``.gmcq-component`` i've invited @guywillis to comment, cos he does a looooot of theming

**Comment by guywillis** [comment]
```
.gmcq-component {
  .graphic-attribution {
    // styles
  }
}
```

I'm happy with this 👍

**Pull request merged**